libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -826,4 +843,33 @@ export class FunctionUtils {
 
         return count;
     }
+
+    /**
+     * @private
+     */
+    private static convertToObj(instance: unknown) {
+        if (FunctionUtils.getPropertyCount(instance) >= 0) {
+            if (instance instanceof Map) {
```

Comment:
Last suggestion, then I promise I won't code-golf anymore on this PR 😄 

```typescript
const entries = instance instanceof Map ? instance : Object.entries(instance);
return entries.reduce((acc, [key, value]) => ({...acc, [key]: FunctionUtils.convertToObj(value)}), {});
```

Reply:
More suggestions better! Means more can learn.
`reduce` could not apply on `IterableIterator` directly. So I use Array.from(instance.entries()) to convert it into Array.